### PR TITLE
feat: custom error page

### PR DESCRIPTION
**Pull request by JStoresund**
This PR is the first draft of a custom error page. The page consists of an outlined box in the middle with some describing text. The text inside the container is made as follows:

1. "Oops"
2. Error code
3. Error message
4. Link to front page

There is one version of the error page for mobile users, and one for computer users. For now they are very similar, only having slightly different text styling. Having one version for each is practical for further improvements if that is needed/wanted.

### CODE REVIEW

File: components/error/errorCardMobile.vue [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    <h2 class="text-h4">Error {{ error.statusCode }}</h2>
+    <p class="text-h6" style="word-break: break-word; max-width: 100%">
+      {{ error.message }}
+    </p>
+    <NuxtLink to="/" class="text-h6">Tilbake til forsiden</NuxtLink>
+  </div>
+</template>
+
+<script setup lang="ts">
+  defineProps(['error'])
```
Comment (JStoresund) [inline components/error/errorCardMobile.vue:22]: Jeg får problemer med at 'error' burde typedefineres. Har forsøk litt ulike ting, men får alltid klager ('error' er av typen Object(), men får ikke til å legge det inn).

File: components/error/errorCard.vue [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<template>
+  <div
+    class="text-center border-1px pa-10"
+    style="
```
Comment (AasmundN) [inline components/error/errorCard.vue:4]: Inline styles like this are fine for only adding one or two attributes. But for adding a lot of styles like this you should make a class and place the styles in the `<style scoped></style>` tag.

File: components/error/errorCardMobile.vue [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    class="text-center border-1px px-10 py-5"
+    style="
+      border: 1px solid black;
+      border-radius: 10px;
+      position: fixed;
+      width: 350px;
+      max-width: 90vw;
+    "
+  >
+    <h1 class="text-h2 font-weight-bold" style="color: orange">Oops!</h1>
```
Comment (AasmundN) [inline components/error/errorCardMobile.vue:12]: I believe Vuetify has utility classes for setting text color as well.

File: components/error/errorCardMobile.vue [hunk 4]
```diff
@@ -0,0 +1,23 @@
+<template>
+  <div
```
Comment (AasmundN) [inline components/error/errorCardMobile.vue:2]: Maybe use the Vuetify `VCard` component in order to have consistent styling with the rest of the application.

**Review by AasmundN — CHANGES_REQUESTED** [review]
Great work so far! I think that the error card component should be one component, no need to separate into one version for mobile and one for desktop. Instead you can use _@media queries_ or the _useDisplay_ composable to dynamically update the styles. However I believe Vuetify component VCard would adjust some its styles automatically out of the box.

File: error.vue [hunk 5]
```diff
@@ -0,0 +1,30 @@
+<template>
+  <div class="d-flex h-screen align-center justify-center">
+    <v-sheet class="text-center">
```
Comment (AasmundN) [inline error.vue:3]: Nice! One thing to take note of is that we always PascalCase for component names. So `VSheet` and not `v-sheet`. Even though both are valid syntax it is smart to keep to established conventions.

File: error.vue [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+      <h4 class="text-h4 pb-1">Error {{ props.error.statusCode }}</h4>
+      <h5 class="text-h5 pb-2 break-text">
+        {{ props.error.message }}
+      </h5>
+      <v-btn @click="handleError">Return to frontpage</v-btn>
+    </v-sheet>
+  </div>
+</template>
+
+<script setup>
```
Comment (AasmundN) [inline error.vue:14]: You will have to add `lang='ts'` in order to use typescript here.

File: pages/[...slug].vue [hunk 7]
```diff
@@ -0,0 +1,18 @@
+<template>
+  <div class="d-flex h-screen align-center justify-center">
+    <v-sheet class="text-center">
+      <h1 class="text-h1 pb-9">404</h1>
+      <h5 class="text-h5 py-3 break-text">
+        Denne siden finnes ikke: /{{ $route.params.slug[0] }}
+      </h5>
+      <v-btn @click="navigateTo('/')">Tilbake til forsiden</v-btn>
```
Comment (AasmundN) [inline pages/[...slug].vue:8]: Remember to localize navigation!

File: pages/[...slug].vue [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+        Denne siden finnes ikke: /{{ $route.params.slug[0] }}
+      </h5>
+      <v-btn @click="navigateTo('/')">Tilbake til forsiden</v-btn>
+    </v-sheet>
+  </div>
+</template>
+
+<style scoped>
+  .break-text {
+    max-width: 90vw;
```
Comment (AasmundN) [inline pages/[...slug].vue:15]: 🤩

**Review by AasmundN — COMMENTED** [review]
Much better! I can seem to get the _error.vue_ page to render, so that will need some looking into. The 404 page is looking smashing! Remember to merge the dev branch into this branch🙌

**Review by JStoresund — COMMENTED** [review]
Most of the changes made in this draft are fixes which were requested from the previous draft. 

A notable problem with the last draft was that upon creating a custom error, the error.vue page didn't show as it should. With some digging into the problem, it is deducable that ***the error.vue page only shows when the error is on the client side***. By adding the code 

´´´
throw createError({
    statusCode: 420,
    statusMessage: 'lmao',
  })
´´´

inside for example _server/middleware/auth.ts_ (or any other typescript file inside the server directory), error.vue does NOT show up.

On the contrary, if the above code is inserted into the script tag in _pages/index.vue_, the error.vue page will indeed be rendered. Hence the conclusion that the error has to be on the client side.

**Review by AasmundN — APPROVED** [review]
This looks good!

**Pull request merged**